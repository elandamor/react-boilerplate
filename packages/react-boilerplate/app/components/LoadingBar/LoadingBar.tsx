import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar />
 */

interface IProps {
  className?: string;
}

const LoadingBar: SFC<IProps> = ({ className }) => (
  <Wrapper className={classNames('c-loadingBar', className)} />
);

LoadingBar.defaultProps = {
  className: '',
};

export default LoadingBar;
