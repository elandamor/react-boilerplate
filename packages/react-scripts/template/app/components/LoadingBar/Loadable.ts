/**
 *
 * Asynchronously loads the component for LoadingBar
 *
 */

import React from 'react';

// @ts-ignore
export default React.lazy(() => import('./index'));
