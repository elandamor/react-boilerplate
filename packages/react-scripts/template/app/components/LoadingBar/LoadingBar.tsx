import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar />
 */

interface IProps {
  className?: string;
  loading?: boolean;
}

const LoadingBar: SFC<IProps> = ({ className, loading }) => (
  <Wrapper
    className={classNames('c-loadingBar', className, {
      '-loading': loading,
    })}
  />
);

export default LoadingBar;
