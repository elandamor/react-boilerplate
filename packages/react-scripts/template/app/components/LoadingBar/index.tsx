import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

export interface IProps {
  className?: string;
  loading?: boolean;
}

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar loading />
 */

const LoadingBar: SFC<IProps> = ({ className, loading }) => (
  <Wrapper
    className={classNames('c-loadingBar', className)}
    {...( loading ? { 'data-loading': true } : { 'data-loading': false } )}
  />
);

export default LoadingBar;
