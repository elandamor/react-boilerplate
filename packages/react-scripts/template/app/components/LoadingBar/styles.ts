import styled, { css, keyframes } from 'styled-components';
import { IProps } from './index';

const shiftRightwards = keyframes`
  0% { transform: translateX(-100%) }
  40% { transform: translateX(0) }
  60% { transform: translateX(0) }
  100% { transform: translateX(100%) }
`;

const Wrapper = styled.div`
  background: var(--brand-success, #000000);
  display: none;
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  transform: translateX(100%);
  z-index: 10;

  ${(props: IProps) =>
    props.loading &&
    css`
      animation: ${shiftRightwards} 1s ease-in-out infinite;
      animation-delay: 0.8s;
      display: block;
    `};
`;

export default Wrapper;
