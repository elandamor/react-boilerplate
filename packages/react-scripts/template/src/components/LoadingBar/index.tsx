import styled, { keyframes } from 'styled-components';

const shiftRightwards = keyframes`
  0% { transform: translateX(-100%) }
  40% { transform: translateX(0) }
  60% { transform: translateX(0) }
  100% { transform: translateX(100%) }
`;

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar />
 */

const LoadingBar = styled.div`
  animation: ${shiftRightwards} 1s ease-in-out infinite;
  animation-delay: 0.8s;
  background: var(--brand-success, #000000);
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  transform: translateX(100%);
  z-index: 10;
`;

export default LoadingBar;
