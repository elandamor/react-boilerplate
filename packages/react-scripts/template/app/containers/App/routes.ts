// tslint:disable:object-literal-sort-keys
import { Home } from '../../pages';

import { IRouteProps } from '../../components/Routes';

const routes: IRouteProps[] = [
  {
    exact: true,
    path: '/',
    component: Home,
  },
];

export default routes;
