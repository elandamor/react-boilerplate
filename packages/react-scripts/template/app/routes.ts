// tslint:disable:object-literal-sort-keys
import { Home, NotFound } from './pages';

import { IRouteProps } from './components/Routes';

const routes: IRouteProps[] = [
  {
    exact: true,
    path: '/',
    component: Home,
  },
  {
    path: '*',
    component: NotFound,
  },
];

export default routes;
