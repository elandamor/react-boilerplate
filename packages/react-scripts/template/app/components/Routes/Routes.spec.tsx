// Routes.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-styled-components';

import routes from '../../containers/App/routes';
import Routes from './index';
import LoadingBar from '../LoadingBar';

describe('<Routes />', () => {
  it('should render Routes without crashing', () => {
    const component = renderer.create(
      <Router>
        <React.Suspense fallback={<LoadingBar loading />}>
          <Routes routes={routes} />        
        </React.Suspense>
      </Router>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});