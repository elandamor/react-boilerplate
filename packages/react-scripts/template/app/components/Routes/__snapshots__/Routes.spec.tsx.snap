// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should render Routes without crashing 1`] = `
.c0 {
  background: var(--brand-success,#000000);
  display: none;
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  z-index: 10;
}

.c0.-loading {
  display: block;
  -webkit-animation: jxImEi 1s ease-in-out infinite;
  animation: jxImEi 1s ease-in-out infinite;
  -webkit-animation-delay: .8s;
  animation-delay: .8s;
}

<div
  className="c-loadingBar -loading c0"
/>
`;
